### PR TITLE
[KernelGen][Nvidia] Add reduce_sum operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a reduce_sum operator that sums elements of a tensor along a dimension or all elements. This is equivalent to torch.sum but implemented as a separate FlagGems operator for extensibility.

## Testing
- Parametrized tests across float16/float32/bfloat16 dtypes (27 cases)
- Validated against PyTorch reference on device side
- Tested on: NVIDIA H20, CUDA 13.2

## Performance
Test command: pytest benchmark/test_reduce_sum.py -m reduce_sum -s --level core (NVIDIA H20)

### float16
| Size Detail | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [1048576] | 0.013 | 0.009 | 1.410 |
| [64, 64] | 0.007 | 0.008 | 0.903 |
| [4096, 4096] | 0.027 | 0.019 | 1.401 |
| [64, 512, 512] | 0.027 | 0.019 | 1.442 |
| [1024, 1024, 1024] | 0.845 | 0.747 | 1.132 |

Avg: **1.258x**

### float32
| Size Detail | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [1048576] | 0.013 | 0.010 | 1.338 |
| [64, 64] | 0.007 | 0.008 | 0.926 |
| [4096, 4096] | 0.035 | 0.029 | 1.206 |
| [64, 512, 512] | 0.035 | 0.029 | 1.210 |
| [1024, 1024, 1024] | 1.170 | 1.770 | 0.661 |

Avg: **1.068x**

### bfloat16
| Size Detail | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [1048576] | 0.012 | 0.009 | 1.388 |
| [64, 64] | 0.007 | 0.008 | 0.911 |
| [4096, 4096] | 0.027 | 0.019 | 1.417 |
| [64, 512, 512] | 0.027 | 0.019 | 1.443 |
| [1024, 1024, 1024] | 0.838 | 0.810 | 1.035 |

Avg: **1.239x**

**Overall: 1.188x**

## Multi-backend Testing
| Backend | Accuracy | Benchmark | Speedup | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS (27/27) | PASS (15 cases) | 1.188 | Primary |
| Tianshu | — | — | — | — |
| Muxi | — | — | — | — |
| Ascend | — | — | — | — |
| Hygon | — | — | — | — |

## Files Changed
- src/flag_gems/ops/reduce_sum.py
- tests/test_reduce_sum.py
- benchmark/test_reduce_sum.py
- src/flag_gems/ops/__init__.py
- src/flag_gems/__init__.py
- conf/operators.yaml

### CODE REVIEW

File: benchmark/test_reduce_sum.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import torch
+
+import flag_gems
+
+from . import base, consts
+
+
+@pytest.mark.reduce_sum
+@pytest.mark.parametrize("dtype", consts.FLOAT_DTYPES)
+def test_reduce_sum_perf(dtype):
```
Comment (w1120029931-bit) [inline benchmark/test_reduce_sum.py:11]: remove perf

File: conf/operators.yaml [hunk 2]
```diff
@@ -1,4 +1,15 @@
 ops:
+  - id: reduce_sum
```
Comment (w1120029931-bit) [inline conf/operators.yaml:2]: inorder

File: tests/test_reduce_sum.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+@pytest.mark.reduce_sum
+@pytest.mark.parametrize("shape", utils.REDUCTION_SHAPES)
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
+def test_reduce_sum_without_dim(shape, dtype):
+    inp = torch.randn(shape, dtype=dtype, device=flag_gems.device)
+    ref_inp = utils.to_reference(inp, True)
+
+    ref_out = torch.sum(ref_inp)
+    with flag_gems.use_gems():
+        res_out = flag_gems.reduce_sum(inp)
```
Comment (w1120029931-bit) [inline tests/test_reduce_sum.py:20]: should be torch op
